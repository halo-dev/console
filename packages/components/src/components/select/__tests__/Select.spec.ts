import { describe, expect, it } from "vitest";
import { VSelect } from "../index";
import { mount } from "@vue/test-utils";

describe("Select", () => {
  it("should render", () => {
    expect(VSelect).toBeDefined();
    expect(mount(VSelect).html()).toMatchSnapshot();
  });

  it("should work with size prop", function () {
    ["lg", "md", "sm", "xs"].forEach((size) => {
      const select = mount(VSelect, { props: { size } });

      expect(select.html()).toMatchSnapshot();
      expect(select.find("select").classes()).toContain(`select-${size}`);
      select.unmount();
    });
  });
});
