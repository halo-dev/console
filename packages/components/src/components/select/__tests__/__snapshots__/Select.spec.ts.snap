// Vitest Snapshot v1

exports[`Select > should render 1`] = `
"<div class=\\"select-wrapper\\"><select class=\\"select-md\\">
    <!--v-if-->
  </select></div>"
`;

exports[`Select > should work with size prop 1`] = `
"<div class=\\"select-wrapper\\"><select class=\\"select-lg\\">
    <!--v-if-->
  </select></div>"
`;

exports[`Select > should work with size prop 2`] = `
"<div class=\\"select-wrapper\\"><select class=\\"select-md\\">
    <!--v-if-->
  </select></div>"
`;

exports[`Select > should work with size prop 3`] = `
"<div class=\\"select-wrapper\\"><select class=\\"select-sm\\">
    <!--v-if-->
  </select></div>"
`;

exports[`Select > should work with size prop 4`] = `
"<div class=\\"select-wrapper\\"><select class=\\"select-xs\\">
    <!--v-if-->
  </select></div>"
`;
